app/routes/changes-enquiries/deferral/functions.js

```diff
@@ -0,0 +1,124 @@
+const i18n = require('i18next');
+const request = require('request-promise');
+
+const dataStore = require('../../../../lib/dataStore');
+const dateHelper = require('../../../../lib/dateHelper');
+const deferralObject = require('../../../../lib/objects/deferralObject');
+const deleteSession = require('../../../../lib/deleteSession');
+const formValidator = require('../../../../lib/formValidator');
+const generalHelper = require('../../../../lib/helpers/general');
+const keyDetailsHelper = require('../../../../lib/keyDetailsHelper');
+const requestHelper = require('../../../../lib/requestHelper');
+
+const root = '/changes-and-enquiries/personal';
+
+const dateRequestReceivedUrl = `${root}/deferral/date-request-received`;
+const defaultDateUrl = `${root}/deferral/default-date`;
+const stopStatePensionUrl = `${root}/stop-state-pension`;
+
+function getDateRequestReceived(req, res) {
+  const awardDetails = dataStore.get(req, 'awardDetails');
+  const details = dataStore.get(req, 'date-request-received', 'deferral');
+  const keyDetails = keyDetailsHelper.formatter(awardDetails);
+  res.render('pages/changes-enquiries/deferral/date-request-received', {
+    backLink: stopStatePensionUrl,
+    details,
+    formAction: dateRequestReceivedUrl,
+    keyDetails,
+  });
+}
+
+function postDateRequestReceived(req, res) {
+  const awardDetails = dataStore.get(req, 'awardDetails');
+  const details = req.body;
+  const errors = formValidator.dateRequestReceived(details);
+  const keyDetails = keyDetailsHelper.formatter(awardDetails);
+  if (Object.keys(errors).length === 0) {
+    dataStore.save(req, 'date-request-received', details, 'deferral');
+    res.redirect(`${root}/deferral/default-date`);
+  } else {
+    res.render('pages/changes-enquiries/deferral/date-request-received', {
+      backLink: stopStatePensionUrl,
+      details,
+      errors,
+      formAction: dateRequestReceivedUrl,
+      keyDetails,
+    });
+  }
+}
+
+function getDefaultDate(req, res) {
+  const awardDetails = dataStore.get(req, 'awardDetails');
+  const defaultDate = dataStore.get(req, 'default-date', 'deferral');
+  const keyDetails = keyDetailsHelper.formatter(awardDetails);
+  const { statePensionDate } = dataStore.get(req, 'awardDetails');
```

Comment:
You've already got `awardDetails` session above. But when you remove `keyDetails` this can stay as is

Reply:
👍